### PR TITLE
fix: MutableClasses: add additional classes

### PR DESCRIPTION
**Pull request by candrews**
Adds a number of additional immutable classes to `MutableClasses.KNOWN_IMMUTABLE_CLASSES`
----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java [hunk 1]
```diff
@@ -15,13 +15,20 @@
...
-            "java.lang.Float", "java.lang.StackTraceElement", "java.math.BigInteger",
+            "java.lang.Short", "java.lang.Boolean", "java.lang.Class", "java.lang.Long", "java.lang.Double",
+            "java.lang.Float", "java.lang.Runtime.Version", "java.lang.StackTraceElement", "java.math.BigInteger",
             "java.math.Decimal", "java.io.File", "java.awt.Font", "java.awt.BasicStroke",
             "java.awt.Color", "java.awt.GradientPaint", "java.awt.LinearGradientPaint",
             "java.awt.RadialGradientPaint", "java.Cursor.", "java.util.UUID", "java.net.URL",
             "java.net.URI", "java.net.Inet4Address", "java.net.Inet6Address", "java.net.InetSocketAddress",
-            "java.security.Permission"));
+            "java.math.BigDecimal", "java.math.BigInteger",
+            "java.security.Permission",
```
Comment (KengoTODA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java:25]: `java.security.Permission` is `abstract` so it's not immutable.

https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/security/Permission.html

**Review by KengoTODA — CHANGES_REQUESTED** [review]

File: spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java (hunk 1 above)
Comment (candrews) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java:25]: This PR does not add `java.security.Permission` - it was there before, added in commit 2e09a8daf5c1dc.

Shall I add a commit to this PR removing it?

**Review by ThrawnCA — DISMISSED** [review]

File: spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java (hunk 1 above)
Comment (candrews) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java]: I added a commit removing `java.security.Permission`

Hopefully this PR can now be merged :crossed_fingers:

**Comment by ThrawnCA** [comment]
@candrews Please try to avoid force-pushing, as it disrupts the commit history and breaks existing links to review changes.

**Comment by candrews** [comment]
> @candrews Please try to avoid force-pushing, as it disrupts the commit history and breaks existing links to review changes.

Sorry about the trouble. I won't force push going forward.

**Review by KengoTODA — CHANGES_REQUESTED** [review]
There are a few conflicts with #1702 and other PRs, please try to resolve them.

**Comment by candrews** [comment]
I believe that this PR is entirely superseded by #1702 and https://github.com/spotbugs/spotbugs/commit/e0966ca82698d78ccb11ede20efdfb5ed773452a

**Pull request closed without merging**